indigo_app/views/works.py

```diff
@@ -165,7 +184,27 @@ def form_valid(self, form):
                     task.locality = self.work.locality
                     task.save()
 
-            # signals
+            work_updated = True
+
+        # update repeals made
```

Comment:
move the validation logic the forms  form_valid method and the forms save method, so validation runs once on the form

Reply:
also overide form has_changed method to check if formset has changed